### PR TITLE
Video clip generation

### PR DESCRIPTION
**Pull request by ale18V**
Modified the code quite a bit so that mpvacious is now able to generate video clips as well as snapshots.
For this reason some configurations were added namely:

- _animated_snapshot_enabled_: boolean value
- _animated_snapshot_fps_: well, the fps of the generated video clip
- __animated_snapshot_width_,animated_snapshot_height_: like the snapshot config
- _animated_snapshot_quality_: an integer between 0 and 100 included

Also added a key binding (ctrl+g) that lets the user change video_clip_enabled at runtime. This is similar to clipboard autocopy.

### Changes to config.lua
As I added some options about video clip generation in the user configurations, I also added validation controls on those parameters.
### Changes to filename_factory.lua
Added a function which generates the name for a video clip. I could not keep using the same one that generates the snapshot filename as it may happen that both snapshot and video clips have ".webp" extension. In that case problems would occur when updating a new note with ctrl+m (in case the user wanted to add both a video clip and a snapshot)
### Changes to encoder.lua

- Added a function that generates video clip in animated webp format.
- Added a state variable (animation_enabled) that can be toggled with ctrl+g, with the respective function. I feel this would have been out of place if put in the main file.
- In order not to pollute the main file (sub2srs) with if/else I made one single function that takes care of video media generation.
- For the same reason I added two functions which take care of generating the filename of video and audio media respectively. In this way the module also takes care of the extensions, without bothering the main.
- Split the interface mainly in two fields: audio and video. I feel like this way it's cleaner.
- Because of the changes made I would suggest renaming this module. I think media_manager would fit its porpouse
### Changes to subs2srs.lua
Basically just modified the functions which exported and updated anki cards, because i changed the interface of the encoder. I tried to leave them as they were before and letting the encoder handle the new addition of the video clip.
Now the filenames are not generated with filename factory but instead, with the encoder interface. The reasons were explained previously.
Also added the key binding ctrl+g for enabling/disabling video clip generation

### CODE REVIEW

File: encoder.lua [hunk 1]
```diff
@@ -152,36 +186,38 @@ mpv.make_audio_args = function(source_path, output_path, start_timestamp, end_ti
...
+
 end
 
+
 local background_play = function(file_path, on_finish)
     return h.subprocess(
-            { 'mpv', '--audio-display=no', '--force-window=no', '--keep-open=no', '--really-quiet', file_path },
-            on_finish
+        { 'mpv', '--audio-display=no', '--force-window=no', '--keep-open=no', '--really-quiet', file_path },
+        on_finish
```
Comment (tatsumoto-ren) [inline encoder.lua:214]: This change in indentation is redundant.

File: subs2srs.lua [hunk 2]
```diff
@@ -342,13 +351,13 @@ local function export_to_anki(gui)
         sub['text'] = string.format("mpvacious wasn't able to grab subtitles (%s)", os.time())
     end
     local snapshot_timestamp = mp.get_property_number("time-pos", 0)
-    local snapshot_filename = filename_factory.make_snapshot_filename(snapshot_timestamp, config.snapshot_extension)
-    local audio_filename = filename_factory.make_audio_filename(sub['start'], sub['end'], config.audio_extension)
+    local video_filename = encoder.video.make_filename(sub['start'], sub['end'], snapshot_timestamp)
+    local audio_filename = encoder.audio.make_filename(sub['start'], sub['end'])
```
Comment (tatsumoto-ren) [inline subs2srs.lua:356]: I don't see why you'd need to move filename_factory to a different module since this doesn't have anything to do with the proposed feature.

File: subs2srs.lua [hunk 3]
```diff
@@ -342,13 +351,13 @@ local function export_to_anki(gui)
...
 
-    encoder.create_snapshot(snapshot_timestamp, snapshot_filename)
-    encoder.create_audio(sub['start'], sub['end'], audio_filename, audio_padding())
+    encoder.video.create(sub['start'], sub['end'], snapshot_timestamp, video_filename)
+    encoder.audio.create(sub['start'], sub['end'], audio_filename, audio_padding())
 
-    local note_fields = construct_note_fields(sub['text'], sub['secondary'], snapshot_filename, audio_filename)
+    local note_fields = construct_note_fields(sub['text'], sub['secondary'], video_filename, audio_filename)
```
Comment (tatsumoto-ren) [inline subs2srs.lua:361]: Technically you're not making any *videos* so `video_filename` and ` encoder.video` are really just `shapshot_filename` and `encoder.snapshot`.

File: encoder.lua [hunk 4]
```diff
@@ -210,15 +246,72 @@ local create_audio = function(start_timestamp, end_timestamp, filename, padding)
...
 local init = function(config, store_fn, platform)
     self.config = config
     self.store_fn = store_fn
     self.platform = platform
     self.encoder = config.use_ffmpeg and ffmpeg or mpv
+
+    self.video.animation_enabled = config.animated_snapshot_enabled
+    self.video.extension = config.animated_snapshot_enabled and config.animated_snapshot_extension or self.config.snapshot_extension
+
+    self.audio.extension = self.config.audio_extension
```
Comment (tatsumoto-ren) [inline encoder.lua:301]: `self.audio.extension` and `self.video.extension` aren't really needed because you're passing their own extensions to each function respectfully.

File: encoder.lua (hunk 1 above)
Comment (ale18V) [reply encoder.lua]: Fixed

File: encoder.lua (hunk 4 above)
Comment (ale18V) [reply encoder.lua:301]: `self.video.extension` is really just equivalent to writing `self.video.animation_enabled and self.config.animated_snapshot_extension or self.config.snapshot_extension`.
The point is it gets updated at runtime whenever animation is toggled on and off, so the conditional expression can be avoided whenever the extension is used in any function (by just writing self.video.extension).
Currently, though the extension is used only in one function, which is the make_video_filename function, so I don't mind replacing self.video.animation_enabled with the conditional expression.

The other one, `self.audio.extension` was added only for the purpose of simmetry, so I'd agree that it's currently useless.

As for `self.video.animation_enabled`, I followed the same pattern used for the clipboard autocopy, namely it gets assigned the config value in the init function and then gets toggled on and off at runtime. I would keep it as it is honestly.

File: subs2srs.lua (hunk 3 above)
Comment (ale18V) [reply subs2srs.lua:361]: Renamed it
